src/Kernels/ModeCouplingKernels.jl

```diff
@@ -16,6 +16,60 @@ struct ModeCouplingKernel3D{F,V,M, M2} <: MemoryKernel
     V3::M
 end
 
+struct dDimModeCouplingKernel{I, F, AF1, AF3} <: MemoryKernel
+    d::I
+    ρ::F
+    kBT::F
+    m::F
+    Nk::I
+    k_array::AF1
+    prefactor::F
+    C::AF1
+    Sk::AF1
+    V::AF3
+    J::AF3
+    P::AF1
+end
+
+function dDimModeCouplingKernel(ρ, kBT, m, k_array, Sₖ, d)
+    Nk = length(k_array)
+
+    Δk = k_array[2] - k_array[1]
+    @assert k_array[1] ≈ Δk / 2
+    @assert all(diff(k_array) .≈ Δk)
+
+    S⁻¹ = inv.(Sₖ)
+    Cₖ = similar(Sₖ)
+
+    for i = 1:Nk
+        Cₖ[i] = (1 - S⁻¹[i]) / ρ
+    end
+
+    prefactor = (kBT / m) * ρ * (Δk)^2 * surface_d_dim_unit_sphere(d-1) / (4*pi)^d
+    P = zeros(Nk)
+    V = zeros(Nk, Nk, Nk)
+    J = zeros(Nk, Nk, Nk)
+
```

Comment:
Make sure all types are correct (this allows measurement errors/autodiff etc)

```suggestion
    Nk = length(k_array)
    
    T = promote_type(eltype(Sₖ), eltype(k_array), typeof(ρ), typeof(kBT), typeof(m))
    ρ, kBT, m = T(ρ), T(kBT), T(m)
    
    k_array, Sₖ = T.(k_array), T.(Sₖ)
    Δk = k_array[2] - k_array[1]
    @assert k_array[1] ≈ Δk / 2
    @assert all(diff(k_array) .≈ Δk)

    S⁻¹ = inv.(Sₖ)
    Cₖ = similar(Sₖ)

    for i = 1:Nk
        Cₖ[i] = (1 - S⁻¹[i]) / ρ
    end

    prefactor = (kBT / m) * ρ * (Δk)^2 * surface_d_dim_unit_sphere(d-1) / (4*pi)^d
    P = zeros(T, Nk)
    V = zeros(T, Nk, Nk, Nk)
    J = zeros(T, Nk, Nk, Nk)

```